src/library/features/crates/cratehierarchy.cpp

```diff
@@ -280,13 +288,34 @@ QString CrateHierarchy::formatQueryForTrackIdsByCratePathLike(const QString& cra
             escapedArgument);
 }
 
-bool CrateHierarchy::nameIsValidForHierarchy(const QString& crateName,
-                                             const CrateId parentId) const {
+bool CrateHierarchy::canBeRenamed(const QString& newName,
+                                  const Crate& crate,
+                                  const CrateId parentId) const {
     if (parentId.isValid()) {
-        return tokenizeCratePath(parentId).contains(crateName);
+        if (collectParentCrateNames(crate).contains(newName) ||
+            collectChildCrateNames(crate).contains(newName)) {
```

Comment:
I'm confused what's going on here. What exactly do `collectParentCrateNames` and `collectChildCrateNames` do? I thought crates would be able to have the same name as another crate as long as the name is unique among crates on the same level (siblings), so why is this walking up and down the hierarchy? Why is this function needed separate from CrateHierarchy::nameIsValidForHierarchy?